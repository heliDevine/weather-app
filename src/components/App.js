import React, { useState, useEffect } from 'react';
import '../styles/App.css';

import LocationDetails from './LocationDetails';
import ForecastSummaries from './ForecastSummaries';
import ForecastDetails from './ForecastDetails';
import getForecast from '../requests/getForecast';
import SearchForm from './SearchForm';

const App = () => {
    const [searchText, setSearchText] = useState('');
    const [forecasts, setForecasts] = useState([]);
    const [location, setLocation] = useState({ city: '', country: '' });
    const [selectedDate, setSelectedDate] = useState(0);
    const selectedForecast = forecasts.find(
        forecast => forecast.date === selectedDate,
    );
    useEffect(() => {
        getForecast(setSelectedDate, setForecasts, setLocation);
    }, []);
    const handleForecastSelect = date => setSelectedDate(date);
    const handleCitySearch = () => {
        getForecast(setSelectedDate, setForecasts, setLocation);
    };
    return (
        <div className="weather-app">
            {/* <h1>Weather app</h1> */}

            <LocationDetails city={location.city} country={location.country} />
            <SearchForm setSearchText />
            <ForecastSummaries
                forecasts={forecasts}
                onForecastSelect={handleForecastSelect}
            />
            {selectedForecast && (
                <ForecastDetails forecast={selectedForecast} />
            )}
        </div>
    );
};

export default App;
