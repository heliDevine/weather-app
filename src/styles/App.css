@import url('https://fonts.googleapis.com/css2?family=Fira+Sans+Condensed:wght@100;400&display=swap');

.weather-app {
    margin: 0 20px;
}
.errorMessage {
    font-family: 'Poiret One', cursive;
    font-weight: 200;
    font-size: 50px;
    color: white;
}
