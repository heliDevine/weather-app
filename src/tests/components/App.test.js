import { render } from '@testing-library/react';
import App from '../../components/App';

describe('Weather app', () => {
    it('renders correctly', () => {
        const { asFragment } = render(<App />);
        expect(asFragment(<App />)).toMatchSnapshot();
    });
});
