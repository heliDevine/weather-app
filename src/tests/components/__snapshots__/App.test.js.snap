// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather app renders correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1>
      WEATHER
    </h1>
    <h2
      class="location-details"
    >
      , 
    </h2>
    <div
      class="search-form"
    >
      <input
        id="location-search"
        placeholder="Enter city"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
         Search 
      </button>
    </div>
    <div
      class="forecast-summaries"
    />
  </div>
</DocumentFragment>
`;
